src/lib.rs

```diff
@@ -1,14 +1,17 @@
 mod constants;
 mod errors;
 mod hash;
+pub mod previous;
```

Comment:
should this be exposed via src/ratchet/previous.rs instead as it's construct behind Ratchet ?

Reply:
Hm yeah right. The module structure is a little weird right now.
I think we actually want to move some things behind `ratchet::*` to the top level, since otherwise you'd be importing `skip_ratchet::ratchet::Ratchet` which.... is kinda redundant :joy:

Reply:
Everywhere usually we just re-export them at lib.rs but we don't really have the mod statatement at lib.rs

Reply:
Yeah, I guess you're referring to `Ratchet::previous`, right? But since that returns `PreviousIterator`, we still need to expose that.
I'm having a hard time understanding what you mean from that sentence :sweat_smile: 

In any case - I restructured the modules at the latest commit, tell me what you think please :)